ui-v2/app/adapters/application.js

```diff
@@ -1,10 +1,23 @@
 import Adapter from './http';
 import { inject as service } from '@ember/service';
+import config from 'consul-ui/config/environment';
 
 export const DATACENTER_QUERY_PARAM = 'dc';
+export const NSPACE_QUERY_PARAM = 'ns';
 export default Adapter.extend({
   repo: service('settings'),
+  nspaceRepo: service('repository/nspace/enabled'),
   client: service('client/http'),
+  formatNspace: function(nspace) {
+    if (config.CONSUL_NSPACES_ENABLED) {
```

Comment:
Does the enterprise binary write this value?

Reply:
It will do at somepoint yeah, I think for now we can just use our CONSUL_BINARY_TYPE setting that we already have - I think this specific NSPACE one is going to be added later. POtentially a PR coming after this one to temporarily use the CONSUL_BINARY_TYPE setting (which the binary writes already)

Reply:
This has been PRed now, see https://github.com/hashicorp/consul/pull/6921 (backend support) and https://github.com/hashicorp/consul/pull/6933 (using these values properly)